File: dataset_reduzido/codenet/Python/Code/s951065247.py
N, A, B, C = map(int, input().split())
l = [int(input()) for i in range(N)]


def dfs(cur, a, b, c):
    if cur == N:
        return abs(A - a) + abs(B - b) + abs(C - c) - 30 if min(a, b, c) > 0 else 10 ** 9
    no_add = dfs(cur + 1, a, b, c)
    add_a = dfs(cur + 1, a + l[cur], b, c) + 10
    add_b = dfs(cur + 1, a, b + l[cur], c) + 10
    add_c = dfs(cur + 1, a, b, c + l[cur]) + 10

    return min(no_add, add_a, add_b, add_c)


print(dfs(0, 0, 0, 0))

File: output/Llama-2-100/codenet/Python/Java/s951065247.java
public class s951065247 {
    public static int dfs(int cur, int a, int b, int c) {
        if (cur == N) {
            return Math.abs(A - a) + Math.abs(B - b) + Math.abs(C - c) - 30;
        }
        int noAdd = dfs(cur + 1, a, b, c);
        int addA = dfs(cur + 1, a + l[cur], b, c) + 10;
        int addB = dfs(cur + 1, a, b + l[cur], c) + 10;
        int addC = dfs(cur + 1, a, b, c + l[cur]) + 10;

        return Math.min(noAdd, addA, addB, addC);
    }

    public static int solve() {
        int N, A, B, C;
        Scanner scanner = new Scanner(System.in);
        N = scanner.nextInt();
        A = scanner.nextInt();
        B = scanner.nextInt();
        C = scanner.nextInt();

        int[] l = new int[N];
        for (int i = 0; i < N; i++) {
            l[i] = scanner.nextInt();
        }

        return dfs(0, 0, 0, 0);
    }
}

